src/main/data_structure/stack.py

```
@@ -0,0 +1,41 @@
+class Stack:
+
+    # Constructor to initiate stack
+    def __init__(self, stack=None):
+        if stack is None:
+            stack = []
+        self.stack = stack
+
+    # this returns the top item of stack
+    def top(self):
+        try:
+            return self.stack[-1]
+        except IndexError:
+            raise
+
+    # pop to remove an item from stack
+    def pop(self):
```

prev code was okay. we just need to throw an exception like

raise Exception('You are trying to pop an empty Stack')
we don't need try catch here